app/src/main/java/org/schabi/newpipe/database/subscription/SubscriptionDAO.kt

```diff
@@ -16,6 +16,13 @@ abstract class SubscriptionDAO : BasicDAO<SubscriptionEntity> {
     @Query("SELECT * FROM subscriptions ORDER BY name COLLATE NOCASE ASC")
     abstract override fun getAll(): Flowable<List<SubscriptionEntity>>
 
+    @Query("""
+        SELECT * FROM subscriptions
+        WHERE name LIKE '%' || :filter || '%'
+        ORDER BY name COLLATE NOCASE ASC
```

Comment:
You made it diacritic sensitive, and it's really annoying for languages with diacritics, because you never know if people have respected diacritics or not.

From [this SO](https://stackoverflow.com/questions/2461522/how-do-i-perform-an-accent-insensitive-compare-e-with-%C3%A8-%C3%A9-%C3%AA-and-%C3%AB-in-sql-ser), collate Latin1_general_CI_AI should do the job
```suggestion
        ORDER BY name COLLATE Latin1_general_CI_AI
```

Reply:
Yep, I did noticed that.

However, I don't think it is that easy to fix it. Android's SQLite doesn't have a collation available for this use case and I'm not sure [if SQLite itself even supports using it for the `LIKE` statement](https://www.sqlite.org/lang_expr.html#like).

Some solutions that I found is to either have a column (in some table) where you normalize these characters when inserting to it, or transforming the search query into a expression that treat those special characters as normal ones (e.g. `[aáàãAÁÀÃ...]`). We could also add support for the accent insensitive `LIKE` natively, but I'm not so sure about that...

Maybe I missed something about this in Android, so let me know if you find some native solution for this.

Still, for the order, `UNICODE` or `LOCALIZED` would've been better here (available in Android).

I'll leave these issues for another PR because that's how it's done in multiple queries across the app, would have to change them as well.